ethereum_history_api/oracles/src/script/prepareNoirFixtures.ts

```diff
@@ -38,35 +51,37 @@ for (const hardFork in FIXTURES) {
   const hardforkModuleFile = `${OUT_DIR}/${hardFork}.nr`;
 
   for (const fixtureName in FIXTURES[hardFork]) {
-    const { blockNumber, address } = FIXTURES[hardFork][fixtureName];
+    const { blockNumber, address, storageKeys } = FIXTURES[hardFork][fixtureName];
 
     const block = await client.getBlock({ blockNumber });
     const stateProof = await client.getProof({
       address,
-      storageKeys: [],
+      storageKeys: storageKeys || [],
       blockNumber
     });
 
-    await mkdir(`${OUT_DIR}/${hardFork}/${fixtureName}`, { recursive: true });
+    const fixturePath = `${OUT_DIR}/${hardFork}/${fixtureName}`;
+    const fixtureFile = (name: string) => `${fixturePath}/${name}.nr`;
 
-    const headerFixture = await createHeaderFixture(block);
-    const headerFile = `${OUT_DIR}/${hardFork}/${fixtureName}/header.nr`;
-    await writeFile(headerFile, headerFixture);
+    await mkdir(fixturePath, { recursive: true });
 
-    const accountFixture = await createAccountFixture(stateProof);
-    const accountFile = `${OUT_DIR}/${hardFork}/${fixtureName}/account.nr`;
-    await writeFile(accountFile, accountFixture);
+    await writeFile(fixtureFile('header'), createHeaderFixture(block));
+    await writeFile(fixtureFile('account'), createAccountFixture(stateProof));
+    await writeFile(fixtureFile('state_proof'), createStateProofFixture(stateProof));
+    if (storageKeys) {
+      await writeFile(
+        fixtureFile('storage_proof'),
+        createStorageProofFixture(stateProof.storageHash, stateProof.storageProof)
+      );
+    }
 
-    const stateProofFixture = await createStateProofFixture(stateProof);
-    const stateProofFile = `${OUT_DIR}/${hardFork}/${fixtureName}/state_proof.nr`;
-    await writeFile(stateProofFile, stateProofFixture);
+    const fixtureModules = ['header', 'account', 'state_proof'];
+    if (storageKeys) {
+      fixtureModules.push('storage_proof');
+    }
 
-    const fixtureModule = `mod header;
-mod account;
-mod state_proof;
-`;
-    const fixtureModuleFile = `${OUT_DIR}/${hardFork}/${fixtureName}.nr`;
-    await writeFile(fixtureModuleFile, fixtureModule);
+    const importFixtureModules = fixtureModules.map((name) => `mod ${name};`).join('\n') + '\n';
+    await writeFile(`${fixturePath}.nr`, importFixtureModules);
```

Comment:
I would propose to do it like this:
```rust
    const modulePath = `${OUT_DIR}/${hardFork}/${fixtureName}`;

    await mkdir(modulePath, { recursive: true });

    await writeFile(join(modulePath, 'header.nr'), createHeaderFixture(block));
    await writeFile(join(modulePath, 'account.nr'), createAccountFixture(stateProof));
    await writeFile(join(modulePath, 'state_proof.nr'), createStateProofFixture(stateProof));
    if (storageKeys) {
      await writeFile(
        join(modulePath, 'storage_proof.nr'),
        createStorageProofFixture(stateProof.storageHash, stateProof.storageProof)
      );
    }

    const fixtureModules = ['header', 'account', 'state_proof'];
    if (storageKeys) {
      fixtureModules.push('storage_proof');
    }

    const importFixtureModules = fixtureModules.map((name) => `mod ${name};`).join('\n') + '\n';
    await writeFile(`${modulePath}.nr`, importFixtureModules);
```

It avoids using the inline lambda. That function abstracted away two things. That noir modules have .nr extension and that we append base path. But still when reading this code - I jumped back to the lambda, cause writing to a file is a low-level operation and it's nice to see - where do I write. Now it's explicit.

Reply:
Here `join` is: `import { join } from 'path';`

Reply:
done